app/models/optimized_image.rb

```diff
@@ -106,7 +106,7 @@ def self.create_for(upload, width, height, opts = {})
 
           # store the optimized image and update its url
           File.open(temp_path) do |file|
-            url = Discourse.store.store_optimized_image(file, thumbnail)
+            url = Discourse.store.store_optimized_image(file, thumbnail, nil, secure: upload.secure?)
```

Comment:
Related to the comment in `base_store.rb`, I don't think we need to pass `nil` here for content type. Simply setting it to `nil` when using the S3 store should suffice.